zmap/zgrab2
# Fixes a few issues with redirects

First, this prevents a DNS lookup from happening when we encounter a redirect, *even if we don't intend to follow it*. This likely addresses some part of #452

Second, if we aren't following redirects, don't have the scan fail in an 'application-error'. We are succeeding in what we intended to do, which is to scan without following redirects

## How to Test

I tested by scanning targets that do redirects both with and without this change, using wireshark to monitor for DNS lookups and reviewing output to ensure that we are the same (within reason for randomized items like cookies and dates)
